### ISSUE TITLE
Multiplayer Campaign save file corruption - d_82

### ISSUE THREAD

**Issue by Hank-Pickles** [Regalis11/Barotrauma#4581]
- [X] I have searched the issue tracker to check if the issue has already been reported.

**Description**
Multiplayer game docked at station after completing a Wreck Dive, game kicked all players to lobby and then failed to load with this error "Barotrauma.Networking.GameClient+<StartGame>d__82"

**Steps To Reproduce**
Custom Sub: "Humpback+++" with a Wreck Dive mission completed. Attached Save file. 

Screenshot : https://imgur.com/a/39rnLt6
Save File : https://drive.google.com/file/d/1BYZNShowh1hl8scjnMEMoQgB_TFMtRFD/view?usp=sharing

**Version**
v0.10.0.9

**Additional information**
We poured many hours into this save and would love to have it back. 

Please forgive me if I have made any mistakes - I searched as much as I could but it was beyond my ability. 

Thank you for your time!

**Comment by juanjp600** [comment]
Should be fixed in https://github.com/Regalis11/Barotrauma-development/commit/b4c55a4cc45f0d926affe0dbe10f73017b362356 and https://github.com/Regalis11/Barotrauma-development/commit/52ac7ea09509ce18c4752c7baba6304a41a92230, until these fixes are pushed in an update here's a patched save file that should load in the current version:
[A&J humpback+++.zip](https://github.com/Regalis11/Barotrauma/files/5740122/A.J.humpback%2B%2B%2B.zip)

**Comment by Hank-Pickles** [comment]
You sir, are an angel, a prince and a scholar.

THANK YOU SO MUCH!

You saved our Christmas :)

Sincerely,

Kyle James- Patrick
Producer
Bear Vs Eagle Films Ltd
www.bearvseagle.com


Cell: 778-708-0272
www.kylejamespatrick.ca

**Comment by Fred-Smashing** [comment]
Run in to what I'm assuming is a related issue.

Attempted a new mission after spending quite some time sorting Inventory and selling items.

On loading the new mission, it kicked everyone from the lobby (except me - the host), but stripped me of all my items.  It also failed to load any of the supplies we'd bought.

Reloading the save causes everyone except the host to be forced into spectator mode, and the same issue of being stripped of items persists.

Server Logs from around the error
https://drive.google.com/drive/folders/1XGlrrD4Iv8dHxMpCIYgxo22WLlEASkwH?usp=sharing

Save File
https://drive.google.com/drive/folders/159AuJ-wUyAYJ_TSWPMOvuo0107o9bh-6?usp=sharing

Hopefully this helps you guys fix this issue, It's pretty sad to lose all that progress to a seemingly random error

**Comment by timgagliardi** [comment]
I think I got the same issue, but we didn't get kicked to the lobby.
Our players where partly loaded, but all items from everyone (including NPCs) were dropped.

After experimenting with the save file, I found out that the issue is the "handheldterminal" inside a metal crate, my character was carrying, while docking.

Removing it from the save file fixed this issue.
@juanjp600 Could you confirm that this is the same bug?

@Fred-Smashing
[GAMER TIME_CharacterData_fixed.zip](https://github.com/Regalis11/Barotrauma/files/5751742/GAMER.TIME_CharacterData_fixed.zip)

This may fixes your game.
I simply delete the entry regarding the handheld terminal.

It looks like this:

```
<Item name="" identifier="handheldterminal" ... >
        <Holdable ... >
        <Terminal ... >
</Item>
```

**Comment by Fred-Smashing** [comment]
> I think I got the same issue, but we didn't get kicked to the lobby.
> Our players where partly loaded, but all items from everyone (including NPCs) were dropped.
> 
> After experimenting with the save file, I found out that the issue is the "handheldterminal" inside a metal crate, my character was carrying, while docking.
> 
> Removing it from the save file fixed this issue.
> @juanjp600 Could you confirm that this is the same bug?
> 
> @Fred-Smashing
> [GAMER TIME_CharacterData_fixed.zip](https://github.com/Regalis11/Barotrauma/files/5751742/GAMER.TIME_CharacterData_fixed.zip)
> 
> This may fixes your game.
> I simply delete the entry regarding the handheld terminal.
> 
> It looks like this:
> 
> ```
> <Item name="" identifier="handheldterminal" ... >
>         <Holdable ... >
>         <Terminal ... >
> </Item>
> ```

You sir are a saviour, that fixed it! 👍

**Comment by juanjp600** [comment]
> 
> Removing it from the save file fixed this issue.
> @juanjp600 Could you confirm that this is the same bug?
> 
This appears to be a different issue, what I fixed earlier was related to connections between map locations. Thanks for helping us figure this one out!

**Comment by Rokvach** [comment]
The included save file doesn't seem to exist anymore, so can't really test this ticket specifically. There was a similar issue posted so I tested the save file included in that ticket (#4626) and didn't spot any issues when loading it. So if the issues are identical then this should be working correctly now as well, closing.
